From 414ef6466187c1452b97b69f946cdd150eaa465d Mon Sep 17 00:00:00 2001
From: Andrew Gregory <andrew.gregory.8@gmail.com>
Date: Mon, 11 Feb 2013 22:13:29 -0500
Subject: [PATCH] disable default -prefix services if MKPREFIX = yes

Certain services were not being enabled by default if PREFIX was set.
This appears to have been intended for Gentoo Prefix systems which are
indicated by MKPREFIX not PREFIX.

Signed-off-by: Andrew Gregory <andrew.gregory.8@gmail.com>
---
 runlevels/Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/runlevels/Makefile b/runlevels/Makefile
index e459d15..7bcf29f 100644
--- a/runlevels/Makefile
+++ b/runlevels/Makefile
@@ -53,7 +53,7 @@ install:
 	if ! test -d "${SYSINITDIR}"; then \
 		${INSTALL} -d ${SYSINITDIR} || exit $$?; \
 		for x in ${SYSINIT}; do \
-			if test -n "${PREFIX}"; then \
+			if test "${MKPREFIX}" = yes; then \
 				grep -q "keyword .*-prefix" ${INITFILES}/"$$x" && continue; \
 			fi; \
 			ln -snf ${INITDIR}/"$$x" ${SYSINITDIR}/"$$x" || exit $$?; done \
@@ -61,7 +61,7 @@ install:
 	if ! test -d "${BOOTDIR}"; then \
 		${INSTALL} -d ${BOOTDIR} || exit $$?; \
 		for x in ${BOOT}; do \
-			if test -n "${PREFIX}"; then \
+			if test "${MKPREFIX}" = yes; then \
 				grep -q "keyword .*-prefix" ${INITFILES}/"$$x" && continue; \
 			fi; \
 			ln -snf ${INITDIR}/"$$x" ${BOOTDIR}/"$$x" || exit $$?; \
@@ -70,7 +70,7 @@ install:
 	if ! test -d "${DEFAULTDIR}"; then \
 		${INSTALL} -d ${DEFAULTDIR} || exit $$?; \
 		for x in ${DEFAULT}; do \
-			if test -n "${PREFIX}"; then \
+			if test "${MKPREFIX}" = yes; then \
 				grep -q "keyword .*-prefix" ${INITFILES}/"$$x" && continue; \
 			fi; \
 			ln -snf ${INITDIR}/"$$x" ${DEFAULTDIR}/"$$x" || exit $$?; done \
@@ -78,7 +78,7 @@ install:
 	if ! test -d "${SHUTDOWNDIR}"; then \
 		${INSTALL} -d ${SHUTDOWNDIR} || exit $$?; \
 		for x in ${SHUTDOWN}; do \
-			if test -n "${PREFIX}"; then \
+			if test "${MKPREFIX}" = yes; then \
 				grep -q "keyword .*-prefix" ${INITFILES}/"$$x" && continue; \
 			fi; \
 			ln -snf ${INITDIR}/"$$x" ${SHUTDOWNDIR}/"$$x" || exit $$?; done \
-- 
1.8.3

